common/metrics/otel_metrics_handler.go

```diff
@@ -147,27 +205,38 @@ func (omp *otelMetricsHandler) Stop(l log.Logger) {
 	omp.provider.Stop(l)
 }
 
-// tagsToAttributes helper to merge registred tags and additional tags converting to attribute.KeyValue struct
-func tagsToAttributes(t1 []Tag, t2 []Tag, e excludeTags) []attribute.KeyValue {
-	var attrs []attribute.KeyValue
+// makeSet returns an otel attribute.Set with the given tags merged with the
+// otelMetricsHandler's tags.
+func (omp *otelMetricsHandler) makeSet(tags []Tag) attribute.Set {
+	if len(tags) == 0 {
+		return omp.set
+	}
+	attrs := make([]attribute.KeyValue, 0, omp.set.Len()+len(tags))
+	for i := omp.set.Iter(); i.Next(); {
+		attrs = append(attrs, i.Attribute())
+	}
+	for _, t := range tags {
+		attrs = append(attrs, omp.convertTag(t))
+	}
+	return attribute.NewSet(attrs...)
+}
 
-	convert := func(tag Tag) attribute.KeyValue {
-		if vals, ok := e[tag.Key()]; ok {
-			if _, ok := vals[tag.Value()]; !ok {
-				return attribute.String(tag.Key(), tagExcludedValue)
-			}
+func (omp *otelMetricsHandler) convertTag(tag Tag) attribute.KeyValue {
```

Comment:
Can you document what this function is doing?  To me, it looks like it's basically converting our internal tag to the otel-specific attribute.KeyValue type, while also censoring tags with specific key/value pairs in our configured excludeTags map. However, I don't really understand why we want to do that last part.